### DISCUSSION TITLE
Making the asset service dynamic

### DISCUSSION THREAD

**Discussion by 0xdef1cafe** [shapeshift/web#1060] [Ideas]
Currently, the asset service provides completely static data. It pulls from multiple sources with a script and we commit the output into the lib repo.

There's been a few requests from product lately that by their nature will mean we'll have to update assets in the redux store dynamically

https://github.com/shapeshift/web/issues/960
https://github.com/shapeshift/web/issues/957

The main issues I foresee doing this is gracefully handling things like tx history with unsupported assets, but this should be fairly easy to test.

IMO we should encapsulate this within the asset service, rather than letting this logic bleed out into web. @cjthompson @natven curious on input here before we process with this. 

Something we may want to consider in conjunction with this is redux persist, to persist the asset data to indexeddb or local storage across page refreshes - otherwise users will have to import custom tokens every time.

**Comment by mrnerdhair** [comment]
I think there's probably a usecase here for using native IndexedDB structures for storing the asset list instead of stringifying everything ahead of time; I'm guessing it would be a lot easier to add/remove assets on the fly if the asset service were backed by an "actual" database.
